packages/grpc-native-core/index.d.ts

```diff
@@ -76,7 +76,7 @@ declare module "grpc" {
    * @param options Options to apply to the loaded file
    * @return The resulting gRPC object
    */
-  export function load(filename: Filename, format?: "proto" | "json", options?: LoadOptions): GrpcObject;
+  export function load<T = GrpcObject>(filename: Filename, format?: "proto" | "json", options?: LoadOptions): T;
```

Comment:
Similar to the issue with `loadObject`, I don't understand what this accomplishes.

Reply:
This way, you can give the return object structure a type. For the "routeguide" example that would be something like this:

```ts
import { Message } from 'protobufjs';
// message types in package "routeguide" defined here:
import * as routeguide from './routeguide';


// a modified copy of Message
interface MessageBase<T> {
	new(values?: Partial<T>, var_args?: string[]): T & Message;
	decode(buffer: ArrayBuffer | ByteBuffer | Buffer | string, length?: number | string, enc?: string): T & Message;
	decodeDelimited(buffer: ByteBuffer | ArrayBuffer | Buffer | string, enc?: string): T & Message;
	decode64(str: string): T & Message;
	decodeHex(str: string): T & Message;
	decodeJSON(str: string): T & Message;
}

// use this as T
export default interface Description {
	routeguide: {
		// types
		Point: MessageBase<routeguide.Point>;
		Rectangle: MessageBase<routeguide.Rectangle>;
		Feature: MessageBase<routeguide.Feature>;
		RouteNote: MessageBase<routeguide.RouteNote>;
		RouteSummary: MessageBase<routeguide.RouteSummary>;
		// services
		RouteGuide: typeof routeguide.RouteGuide.Client;
	}
}
```

This will give you a type checked object you can use later on.